### PR TITLE
Yeet `QueryTypeRelatingDelegate`, move `NllTypeRelating` into borrowck crate

### PR DESCRIPTION
**Pull request by compiler-errors**
We can just use the existing equate relation for query instantiation. I don't expect us to want to move everything into `TypeRelating`.

r? lcnr

### CODE REVIEW

File: compiler/rustc_borrowck/src/type_check/relate_tys.rs [hunk 1]
```diff
@@ -84,7 +84,7 @@ impl<'me, 'bccx, 'tcx> NllTypeRelatingDelegate<'me, 'bccx, 'tcx> {
     }
 }
 
-impl<'tcx> TypeRelatingDelegate<'tcx> for NllTypeRelatingDelegate<'_, '_, 'tcx> {
+impl<'tcx> NllTypeRelatingDelegate<'tcx> for NllNllTypeRelatingDelegate<'_, '_, 'tcx> {
```
Comment (lcnr) [inline compiler/rustc_borrowck/src/type_check/relate_tys.rs:87]: this is now the only `NllNllTypeRelatingDelegate` in existance, just stop using a delegate and inline this into the relation? That does require moving the whole relation to `rustc_borrowck` which... does that cause some privacy issues?

Also... one `Nll` to many here

i am in generally in favor of this change

Comment (compiler-errors) [reply compiler/rustc_borrowck/src/type_check/relate_tys.rs]: NllNll was a typo, lol

I just inlined the whole thing into borrowck

File: compiler/rustc_borrowck/src/type_check/relate_tys.rs [hunk 2]
```diff
@@ -163,11 +315,260 @@ impl<'tcx> TypeRelatingDelegate<'tcx> for NllTypeRelatingDelegate<'_, '_, 'tcx>
...
+        let a = self.type_checker.infcx.shallow_resolve(a);
+        assert!(!b.has_non_region_infer(), "unexpected inference var {:?}", b);
+
+        match b.kind() {
+            ty::ConstKind::Infer(ty::InferConst::Var(_)) => {
+                // Forbid inference variables in the RHS.
+                self.type_checker
+                    .infcx
+                    .dcx()
+                    .span_delayed_bug(self.span(), format!("unexpected inference var {b:?}",));
```
Comment (lcnr) [inline compiler/rustc_borrowck/src/type_check/relate_tys.rs:447]: existing, but can you convert this to a bug, the same as for types?

File: compiler/rustc_borrowck/src/type_check/relate_tys.rs [hunk 3]
```diff
@@ -180,4 +581,34 @@ impl<'tcx> TypeRelatingDelegate<'tcx> for NllTypeRelatingDelegate<'_, '_, 'tcx>
...
+                a.into(),
+                ty::AliasRelationDirection::Subtype,
+            ),
+            ty::Variance::Invariant => ty::PredicateKind::AliasRelate(
+                a.into(),
+                b.into(),
+                ty::AliasRelationDirection::Equate,
+            ),
+            // FIXME(deferred_projection_equality): Implement this when we trigger it.
+            // Probably just need to do nothing here.
```
Comment (lcnr) [inline compiler/rustc_borrowck/src/type_check/relate_tys.rs:608]: should be unreachable

**Review by lcnr — COMMENTED** [review]
two existing nits, then r=me

**Pull request merged**